promotion/service.go

```diff
@@ -13,6 +19,36 @@ type Service struct {
 	ledgerClient     ledger.Client
 	reputationClient reputation.Client
 	eventChannel     chan []byte
+	suggestionCodec  *goavro.Codec
```

Comment:
i renamed this to `suggestionCodec` but what do you think about having a codec dictionary per topic? ala `codecs map[string]*goavro.Code`